knimin/lib/qiita_jira_util.py

```diff
@@ -1,41 +1,86 @@
-#!/usr/bin/env python
-from knimin import jira_handler
-from jira import JIRAError
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The LabAdmin Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
 
+from knimin import jira_handler, qiita_client, db
 
-def create_project(project_name, assignee=None,
-                   template_name="Task management"):
-    """Returns an open JIRA connection handler
+
+def create_study(title, abstract, description, alias, qiita_user,
+                 qiita_pi, qiita_lp, jira_user,
+                 jira_template="Task management"):
+    """Creates a study in Qiita, Jira and LabAdmin
 
     Parameters
     ----------
-    project_name : str
-        The project name
-    assignee : str, optional
-        Username of the person assigned to this project
-    template_name : str, optional
-        The template name
+    title: str
+        The study title
+    abstract: str
+        The study abstract
+    description : str
+        The study description
+    alias: str
+        The study alias
+    qiita_user : str
+        The Qiita user that will own the study
+    qiita_pi : dict of str
+        The Qiita study Principal Investigator name, affiliation,
+        and (optionally) email
+    qiita_lp : dict of str
+        The Qiita study lab person name, affiliation and (optionally) email
+    jira_user : str
+        The JIRA user that will lead the project
+    jira_template : str, optional
+        The JIRA template name
 
     Returns
     -------
-    jira.resources.Project
-        The recently created JIRA project
-    str
-        Error message if failure
+    int
+        The study ID
+
+    Raises
+    ------
+    ValueError
+        If there is any problem creating the Qiita person or the Qiita study
     """
-    # study_id will be provided by Qiita, hardcoding it
-    study_id = 10001
-    message = ''
+    # Step 1: Create the study in Qiita
+    # Make sure that the qiita_pi and qiita_lp exist in Qiita
+    # We will assume that if the email is given, then the person needs to be
+    # created in qiita
+    for val in [qiita_pi, qiita_lp]:
+        if 'email' in val:
+            status_code, msg = qiita_client.post('/api/v1/person', data=val)
```

Comment:
If the person already exists the [method returns a 409](https://github.com/biocore/qiita/blob/master/qiita_pet/handlers/rest/study_person.py#L41)

The way that I had it here is that only the name and affiliation is required (just as in Qiita). In case that the user wants to create a new user, then the email is also provided. Does it make sense?